usecases/cluster/transactions_read.go

```diff
@@ -0,0 +1,53 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+func (c *TxManager) CloseReadTransaction(ctx context.Context,
+	tx *Transaction,
+) error {
+	c.Lock()
+	if c.currentTransaction == nil || c.currentTransaction.ID != tx.ID {
+		c.Unlock()
+		return ErrInvalidTransaction
+	}
+
+	c.Unlock()
+
+	// now that we know we are dealing with a valid transaction: no  matter the
+	// outcome, after this call, we should not have a local transaction anymore
+	defer func() {
+		c.Lock()
+		c.currentTransaction = nil
+		c.Unlock()
+	}()
+
+	if err := c.remote.BroadcastCommitTransaction(ctx, tx); err != nil {
+		// we could not open the transaction on every node, therefore we need to
+		// abort it everywhere.
+
+		if err := c.remote.BroadcastAbortTransaction(ctx, tx); err != nil {
+			// TODO WARN with structured logging
```

Comment:
It is not clear to me if you want to handle this TODO in this PR.

Reply:
This was copy/pasted from another place where it was already there. But now is as good a time as any. I'll add structured logging here and in the write transaction where I copied it from.

Reply:
Fixed.